server/src/apub/shared_inbox.rs

```diff
@@ -0,0 +1,222 @@
+use super::*;
+
+#[serde(untagged)]
+#[derive(Serialize, Deserialize, Debug)]
+pub enum SharedAcceptedObjects {
+  Create(Create),
+  Update(Update),
+}
+
+impl SharedAcceptedObjects {
+  fn object(&self) -> Option<&BaseBox> {
+    match self {
+      SharedAcceptedObjects::Create(c) => c.create_props.get_object_base_box(),
+      SharedAcceptedObjects::Update(u) => u.update_props.get_object_base_box(),
+    }
+  }
+}
+
+/// Handler for all incoming activities to user inboxes.
+pub async fn shared_inbox(
+  request: HttpRequest,
+  input: web::Json<SharedAcceptedObjects>,
+  db: DbPoolParam,
+  chat_server: ChatServerParam,
+) -> Result<HttpResponse, Error> {
+  // TODO: would be nice if we could do the signature check here, but we cant access the actor property
```

Comment:
@asonix Any chance to rearrange the activities so that `actor` can be retrieved without figuring out the exact type of activity?

Reply:
that `SharedAcceptedObjects` is a type y'all made. Take a look at the `fn object(&self)` method on it. You can do the same for `actor`